### PR TITLE
Add unimplemented ChainReader() method to MedianProvider

### PR DESCRIPTION
**Pull request by reductionista**
Add ChainReader() method to medianProvider to satisfy the updated PluginProvider interface which we plan to commit soon to chainlink-relay:

https://github.com/smartcontractkit/chainlink-relay/pull/196

### CODE REVIEW

File: relayer/go.mod [hunk 1]
```diff
@@ -91,5 +91,8 @@ replace (
 	github.com/hashicorp/go-plugin => github.com/smartcontractkit/go-plugin v0.0.0-20231003134350-e49dad63b306
 
 	// until merged upstream: https://github.com/mwitkow/grpc-proxy/pull/69
+
```
Comment (jmank88) [inline relayer/go.mod]:
```suggestion
```

**Comment by jmank88** [comment]
It looks like there are some linter failures occurring. Are all of the `go.mod` files tidy?

**Comment by ilija42** [comment]
> It looks like there are some linter failures occurring. Are all of the `go.mod` files tidy?

Tidy doesn't seem to clear these up, also looks proper in local?

```
  <file name="common/common.go">
    <error column="33" line="222" message="undefined: null" severity="error" source="typecheck"></error>
    <error column="33" line="223" message="undefined: null" severity="error" source="typecheck"></error>
  </file>

  <file name="devnet/devnet.go">
    <error column="12" line="20" message="undefined: resty" severity="error" source="typecheck"></error>
    <error column="13" line="29" message="undefined: resty" severity="error" source="typecheck"></error>
  </file>
```

File: ops/go.sum
Comment (jmank88) [inline ops/go.sum:1]: Why is this deleted?

Comment (jmank88) [reply ops/go.sum:1]: The root level make command should tidy everything correctly for you.

Comment (ilija42) [reply ops/go.sum:1]: probably accidental. Restored it now

File: integration-tests/go.mod [hunk 2]
```diff
@@ -8,20 +8,20 @@ require (
...
 	github.com/satori/go.uuid v1.2.1-0.20181028125025-b2ce2384e17b
 	github.com/smartcontractkit/caigo v0.0.0-20230621050857-b29a4ca8c704
-	github.com/smartcontractkit/chainlink-common v0.1.7-0.20231117021201-6814387d8d3e
+	github.com/smartcontractkit/chainlink-common v0.1.7-0.20231123003013-379db0b9e4c7
 	github.com/smartcontractkit/chainlink-starknet/ops v0.0.0-20230329050701-40e3b18bb026
-	github.com/smartcontractkit/chainlink-starknet/relayer v0.0.1-beta-test.0.20231116220530-9d5db8010d3b
+	github.com/smartcontractkit/chainlink-starknet/relayer v0.0.1-beta-test.0.20231117204155-b253a2f56664
 	github.com/smartcontractkit/chainlink/integration-tests v0.0.0-20231116224343-7780eb63d8dc
-	github.com/smartcontractkit/chainlink/v2 v2.5.1-0.20231116224343-7780eb63d8dc
+	github.com/smartcontractkit/chainlink/v2 v2.5.1-0.20231122111232-f7c254bc9c73
```
Comment (jmank88) [inline integration-tests/go.mod]: We typically keep these two aligned with the same hash, since integration-tests uses a relative reference to chainlink/v2.
Let's update `integration-tests` to f7c254bc9c73

**Review by jmank88 — APPROVED** [review]
